plugin/trino-bigquery/src/main/java/io/trino/plugin/bigquery/BigQueryClient.java

```diff
@@ -159,7 +235,72 @@ String selectSql(TableId table, String formattedColumns)
 
     private String fullTableName(TableId tableId)
     {
-        tableId = tableIds.getOrDefault(tableId, tableId);
+        String remoteSchemaName = toRemoteDataset(tableId.getProject(), tableId.getDataset())
+                .map(RemoteDatabaseObject::getOnlyRemoteName)
+                .orElse(tableId.getDataset());
+        String remoteTableName = toRemoteTable(tableId.getProject(), remoteSchemaName, tableId.getTable())
+                .map(RemoteDatabaseObject::getOnlyRemoteName)
+                .orElse(tableId.getTable());
+        tableId = TableId.of(tableId.getProject(), remoteSchemaName, remoteTableName);
         return format("%s.%s.%s", tableId.getProject(), tableId.getDataset(), tableId.getTable());
     }
+
+    List<BigQueryColumnHandle> getColumns(BigQueryTableHandle tableHandle)
+    {
+        TableInfo tableInfo = getTable(tableHandle.getRemoteTableName().toTableId());
+        if (tableInfo == null) {
+            throw new TableNotFoundException(
+                    tableHandle.getSchemaTableName(),
+                    format("Table '%s' not found", tableHandle.getSchemaTableName()));
+        }
+        Schema schema = tableInfo.getDefinition().getSchema();
+        if (schema == null) {
+            throw new TableNotFoundException(
+                    tableHandle.getSchemaTableName(),
+                    format("Table '%s' has no schema", tableHandle.getSchemaTableName()));
```

Comment:
when is this code triggered?

Reply:
Not sure, but the BigQuery client marks `getSchema` as `@Nullable`.

Reply:
I'm adding the annotation here to make sure it's clear and we don't miss checking it.